### PR TITLE
[LMS-259] Configure different directory for the sync database

### PR DESCRIPTION
**Pull request by lalrg**

### CODE REVIEW

File: src/canvas-extractor/README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ Supported parameters:
 | Base Canvas URL | yes | `-b` or `--base-url` | CANVAS_BASE_URL |
 | Canvas API access token | yes | `-a` or `--access-token` | CANVAS_ACCESS_TOKEN |
 | Output Directory | no (default: [working directory]/data) | `-o` or `--output-directory` | OUTPUT_DIRECTORY |
+| Sync database directory | no (default: ./data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```
Comment (stephenfuqua) [inline src/canvas-extractor/README.md:67]: For consistency with the line above:

```suggestion
| Sync database directory | no (default: [working directory]/data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```

File: src/canvas-extractor/edfi_canvas_extractor/config.py [hunk 2]
```diff
@@ -39,14 +37,14 @@ def get_sync_db_engine() -> sqlalchemy.engine.base.Engine:
     """
     running_in_notebook: bool = _is_running_in_notebook()
     logger.debug("Running in Jupyter Notebook: %s", running_in_notebook)
-    sync_database_directory = (
-        os.path.join("..", SYNC_DATABASE_LOCATION_SUFFIX)
+    sync_database_path = (
+        os.path.join("..", sync_database_directory)
```
Comment (stephenfuqua) [inline src/canvas-extractor/edfi_canvas_extractor/config.py:41]: Why join to the parent directory?

File: src/google-classroom-extractor/README.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ Supported parameters:
 | The email address of the Google Classroom admin account. | yes | `-a` or `--classroom-account` | CLASSROOM_ACCOUNT |
 | The log level for the tool. | no (default: INFO) | `-l` or `--log-level` | LOG_LEVEL |
 | The output directory for the generated csv files. | no (default: data/) | `-s` or `--usage-start-date` | OUTPUT_PATH |
+| Sync database directory | no (default: ./data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```
Comment (stephenfuqua) [inline src/google-classroom-extractor/README.md:67]:
```suggestion
| The output directory for the generated csv files. | no (default: [working directory]/data/) | `-s` or `--usage-start-date` | OUTPUT_PATH |
| Sync database directory | no (default: [working directory]/data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```

File: src/schoology-extractor/README.md [hunk 4]
```diff
@@ -72,6 +72,7 @@ to `.env`, and adjust to your desired parameters. Supported parameters:
 | Schoology API Secret | yes | `-s` or `--client-secret` | SCHOOLOGY_SECRET |
 | Usage analytics input directory | no | `-i` or `--input-directory` | SCHOOLOGY_INPUT_DIRECTORY |
 | Output Directory | no (default: [working directory]/data) | `-o` or `--output-directory` | SCHOOLOGY_OUTPUT_PATH |
+| Sync database directory | no (default: ./data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```
Comment (stephenfuqua) [inline src/schoology-extractor/README.md:75]:
```suggestion
| Sync database directory | no (default:  [working directory]/data) | `-d` or `--sync-database-directory` | SYNC_DATABASE_DIRECTORY |
```

File: src/schoology-extractor/edfi_schoology_extractor/helpers/sync.py [hunk 5]
```diff
@@ -29,13 +28,13 @@ def get_sync_db_engine() -> sqlalchemy.engine.base.Engine:
     sqlalchemy.engine.base.Engine
         a SQL Alchemy Engine
     """
-    sync_database_directory = os.path.join(SYNC_DATABASE_LOCATION_SUFFIX)
+    sync_database_path = os.path.join(sync_database_directory)
```
Comment (stephenfuqua) [inline src/schoology-extractor/edfi_schoology_extractor/helpers/sync.py:31]: Seems like `sync_database_path` isn't actually needed.

File: src/google-classroom-extractor/edfi_google_classroom_extractor/config.py [hunk 6]
```diff
@@ -39,14 +38,14 @@ def get_sync_db_engine() -> sqlalchemy.engine.base.Engine:
     """
     running_in_notebook: bool = _is_running_in_notebook()
     logger.debug("Running in Jupyter Notebook: %s", running_in_notebook)
-    sync_database_directory = (
-        os.path.join("..", SYNC_DATABASE_LOCATION_SUFFIX)
+    sync_database_path = (
+        os.path.join("..", sync_database_directory)
```
Comment (stephenfuqua) [inline src/google-classroom-extractor/edfi_google_classroom_extractor/config.py:42]: Again, wondering why trying to make this relative to the parent directory. Also what happens if `sync_database_directory` is a full path already, not a relative path?

**Review by stephenfuqua — CHANGES_REQUESTED** [review]

File: src/google-classroom-extractor/edfi_google_classroom_extractor/config.py (hunk 6 above)
Comment (lalrg) [reply src/google-classroom-extractor/edfi_google_classroom_extractor/config.py:42]: It only happens when is running in a notebook, I don't remember why this validation was added. 🤔

File: src/canvas-extractor/edfi_canvas_extractor/config.py (hunk 2 above)
Comment (lalrg) [reply src/canvas-extractor/edfi_canvas_extractor/config.py]: Fixed ✔️

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**